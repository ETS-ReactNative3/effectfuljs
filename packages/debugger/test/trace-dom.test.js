require("./setupTrace");
const trace = require("./run").bind(null, true);
const D = require("../main");

test("DOM time traveling", function() {
  global.console = { log: jest.fn() };
  document.body.innerHTML = `<div id="root"></div>`;
  D.evalThunk(require("./__fixtures__/dom"));
  const mod = trace();
  const c1 = D.Trace.checkpoint();
  D.Trace.DOM.track(document.getElementById("root"));
  mod.changeDom1();
  trace(true);
  expect(document.body).toMatchSnapshot();
  D.Trace.checkpoint();
  mod.changeDom2();
  trace(true);
  expect(document.body).toMatchSnapshot();
  const c2 = D.Trace.checkpoint();
  mod.changeDom3();
  trace(true);
  expect(document.body).toMatchSnapshot();
  const c3 = D.Trace.checkpoint();
  mod.changeDom4();
  trace(true);
  expect(document.body).toMatchSnapshot();
  D.Trace.undo(c3);
  expect(document.body).toMatchSnapshot();
  mod.changeDom4();
  trace(true);
  expect(document.body).toMatchSnapshot();
  D.Trace.undo(c2);
  expect(document.body).toMatchSnapshot();
  mod.changeDom3();
  trace(true);
  expect(document.body).toMatchSnapshot();
  mod.changeDom4();
  trace(true);
  expect(document.body).toMatchSnapshot();
  D.Trace.undo(c1);
  expect(document.body).toMatchSnapshot();
});
