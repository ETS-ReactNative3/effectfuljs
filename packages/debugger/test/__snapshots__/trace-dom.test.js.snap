// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOM time traveling 1`] = `
<body>
  <div
    id="root"
  >
    <div
      id="i1"
      style="color:green"
    >
      Hi
    </div>
  </div>
</body>
`;

exports[`DOM time traveling 2`] = `
<body>
  <div
    id="root"
  >
    <div
      id="i1"
      style="color:red"
    >
      <div
        id="i2"
      >
        There
      </div>
    </div>
  </div>
</body>
`;

exports[`DOM time traveling 3`] = `
<body>
  <div
    id="root"
  >
    <div
      id="i1"
      style="color:blue"
    >
      <h1
        id="i3"
      />
      <div
        id="i2"
      >
        There
      </div>
      <h2
        id="i4"
      />
    </div>
  </div>
</body>
`;

exports[`DOM time traveling 4`] = `
<body>
  <div
    id="root"
  >
    <div
      id="i1"
      style="color:blue"
    >
      <h1
        id="i3"
      />
      <div
        id="i2"
      >
        THERE
      </div>
    </div>
  </div>
</body>
`;

exports[`DOM time traveling 5`] = `
<body>
  <div
    id="root"
  >
    <div
      id="i1"
      style="color:blue"
    >
      <h1
        id="i3"
      />
      <div
        id="i2"
      >
        There
      </div>
      <h2
        id="i4"
      />
    </div>
  </div>
</body>
`;

exports[`DOM time traveling 6`] = `
<body>
  <div
    id="root"
  >
    <div
      id="i1"
      style="color:blue"
    >
      <h1
        id="i3"
      />
      <div
        id="i2"
      >
        THERE
      </div>
    </div>
  </div>
</body>
`;

exports[`DOM time traveling 7`] = `
<body>
  <div
    id="root"
  >
    <div
      id="i1"
      style="color:red"
    >
      <div
        id="i2"
      >
        There
      </div>
    </div>
  </div>
</body>
`;

exports[`DOM time traveling 8`] = `
<body>
  <div
    id="root"
  >
    <div
      id="i1"
      style="color:blue"
    >
      <h1
        id="i3"
      />
      <div
        id="i2"
      >
        There
      </div>
      <h2
        id="i4"
      />
    </div>
  </div>
</body>
`;

exports[`DOM time traveling 9`] = `
<body>
  <div
    id="root"
  >
    <div
      id="i1"
      style="color:blue"
    >
      <h1
        id="i3"
      />
      <div
        id="i2"
      >
        THERE
      </div>
    </div>
  </div>
</body>
`;

exports[`DOM time traveling 10`] = `
<body>
  <div
    id="root"
  />
</body>
`;
